.github/workflows/publish-prerelease.yml

```diff
@@ -32,25 +32,30 @@ jobs:
           REPOSITORY: ${{ github.event.repository.name }}
-          # For a `pull_request` event, the branch is `github.head_ref``.
-          BRANCH: ${{ github.head_ref }}
+          # For a `pull_request` event, the fork is `github.event.pull_request.head.repo.fork`, and the branch is `github.head_ref`.
+          BRANCH: ${{ github.event.pull_request.head.repo.fork && format('pull/{0}', github.event.pull_request.number) || github.head_ref }}
           # For a `pull_request` event, the head commit hash is `github.event.pull_request.head.sha`.
           HEAD_COMMIT_HASH: ${{ github.event.pull_request.head.sha }}
           JOB_NAME: job-publish-prerelease
         run: |
-          pipeline_id=$(curl --silent "https://circleci.com/api/v2/project/gh/$OWNER/$REPOSITORY/pipeline?branch=$BRANCH" | jq --arg head_commit_hash "${HEAD_COMMIT_HASH}" -r '.items | map(select(.vcs.revision == $head_commit_hash)) | first | .id')
-          workflow_id=$(curl --silent "https://circleci.com/api/v2/pipeline/$pipeline_id/workflow" | jq -r ".items[0].id")
-          job_details=$(curl --silent "https://circleci.com/api/v2/workflow/$workflow_id/job" | jq --arg job_name "${JOB_NAME}" -r '.items[] | select(.name == $job_name)')
+          pipeline=$(curl --silent "https://circleci.com/api/v2/project/gh/${OWNER}/${REPOSITORY}/pipeline?branch=${BRANCH}" | jq --arg head_commit_hash "${HEAD_COMMIT_HASH}" -r '.items | map(select(.vcs.revision == $head_commit_hash)) | first')
+          pipeline_id=$(echo "${pipeline}" | jq -r '.id')
+          pipeline_number=$(echo "${pipeline}" | jq -r '.number')
 
-          build_num=$(echo "$job_details" | jq -r '.job_number')
-          echo 'CIRCLE_BUILD_NUM='"$build_num" >> "$GITHUB_OUTPUT"
+          workflow=$(curl --silent "https://circleci.com/api/v2/pipeline/${pipeline_id}/workflow" | jq -r '.items[0]')
+          workflow_id=$(echo "${workflow}" | jq -r '.id')
 
-          job_id=$(echo "$job_details" | jq -r '.id')
-          echo 'CIRCLE_WORKFLOW_JOB_ID='"$job_id" >> "$GITHUB_OUTPUT"
+          job=$(curl --silent "https://circleci.com/api/v2/workflow/${workflow_id}/job" | jq --arg job_name "${JOB_NAME}" -r '.items[] | select(.name == $job_name)')
+          job_id=$(echo "${job}" | jq -r '.id')
+          job_number=$(echo "${job}" | jq -r '.job_number')
 
-          echo "Getting artifacts from pipeline '${pipeline_id}', workflow '${workflow_id}', build number '${build_num}', job id '${job_id}'"
+          echo "CIRCLE_WORKFLOW_JOB_ID=${job_id}" >> "$GITHUB_OUTPUT"
+          echo "CIRCLE_BUILD_NUM=${job_number}" >> "$GITHUB_OUTPUT"
+
+          echo "Getting artifacts from pipeline number '${pipeline_number}', workflow id '${workflow_id}', job number '${job_number}' at https://app.circleci.com/pipelines/github/${OWNER}/${REPOSITORY}/${pipeline_number}/workflows/${workflow_id}/jobs/${job_number}/artifacts"
 
       - name: Publish prerelease
+        if: ${{ env.PR_COMMENT_TOKEN && vars.AWS_CLOUDFRONT_URL }}
```

Comment:
whats the reason/difference between checking `vars.AWS_CLOUDFRONT_URL` vs `vars.AWS_REGION && vars.AWS_IAM_ROLE && vars.AWS_S3_BUCKET `?